### ISSUE TITLE
How to make a custom logger?

### ISSUE THREAD

**Issue by dhiraj** [sushichop/Puppy#21]
I'd like to make a custom logger that would log to a new target, specifically in my case Firebase Crashlytics. Looking at `ConsoleLogger` which inherits from `BaseLogger`, I would think that all I needed to do would be to define my own `CrashlyticsLogger` with an overridden `log()` function.

However, since the `BaseLogger` class is defined as `public` instead of `open` I'm not able to do so, with the compiler error: `Cannot inherit from non-open class 'ConsoleLogger' outside of its defining module`.

Is there some other mechanism that I should be using to make custom loggers? 🤔

**Comment by sushichop** [comment]
@dhiraj 

Thanks for letting me know. You are right.
I've changed the accessor from `public` to `open` and released a new version. [Puppy 0.3.0](https://github.com/sushichop/Puppy/releases/tag/0.3.0)
Please use this version.

To create a custom logger, you just need to inherit `BaseLogger` class and override `log(_:string:)` method.
And I have added this information to the last part of the README.

```swift
import Puppy

class CustomLogger: BaseLogger {
    override func log(_ level: LogLevel, string: String) {
        // Implements the logging feature here.
    }
}
```

Thanks.

**Comment by sushichop** [comment]
I'm closing this issue since it has been fixed by #23 and [Puppy 0.3.0](https://github.com/sushichop/Puppy/releases/tag/0.3.0).

Thanks.
